### ISSUE TITLE
Bug with saved cookies restoration

### ISSUE THREAD

**Issue by vorukyu** [Phatsuo/awesome-cookie-manager#5]
I've encountered a bug in the latest version of google chrome: when you try to restore saved cookies, the ones that have dot as a first character in domain name are not restored.
My quick fix was to remove this preceding dot from the domain in created URL in restoreCookies(), and also to add this domain in cookies.set() if hostOnly == true, but maybe the last change is unnecessary.

Could you take a look at this?

**Comment by Procyon-b** [comment]
I' haven't paid attention to it. I never restore cookies (at least not one that falls into the category of this new problem), but I had to do it a couple of days ago while helping a family member migrating to a new computer.
The solution I've used is to write a small scriplet to generate `document.cookie="..."` lines. But it's not feasable in an everyday situation.

I've missed your message. Sorry. Apparently @Phatsuo hasn't noticed it either.
I'll look into it and try to find a solution.
I wonder what has been changed in chrome's code to block this.

**Comment by Procyon-b** [comment]
@vorukyu @Phatsuo 
Good news. I've found what is causing the error!   :)
I still have to clean the code. (The dev version I'm using has other modifications)
I'll make a PR later today or tomorrow.

In a nutshell, when setting a `.domain` type cookie, the `domain` has to be set, and the `url` must not contain a leading dot.

vorukyo, maybe that's exactly what you did. After re-reading your explanation it seems clearer now.  ;)

**Comment by vorukyu** [comment]
@Procyon-b yeah, my explanation looks too confusing even for me now 😀 Thanks for looking into the issue.

**Comment by Procyon-b** [comment]
@vorukyu I'm waiting for Phatsuo reaction (I emailed him). I know it can take a while.
I'm taking the occasion of this new release to modify little things.
Your input is welcome:
- cookie sets saved in incognito mode are now identified
![ACM_incoSave](https://user-images.githubusercontent.com/40043075/81567169-3a44f380-939c-11ea-89f7-413b2d61a198.PNG)
but, while testing I noticed that cookies from incognito mode can be restored from a non-incognito tab, and cookie from normal mode can be restored from an incognito tab. I think it started when chrome's behaviour with cookieStores changed a year and a half ago (when I PR-ed a fix).
It can go unnoticed since these cookies won't appear in the listing in the popup (other cookieStore). 

- The orange dot can help here, but what do you think of preventing the user of restoring cookies in one mode from the other mode?

- this can be option driven. Phatsuo will have the last say on this, but I can propose it in the PR.
